test/dsl/dsl_model_construction.jl

```diff
@@ -367,3 +367,32 @@ rn = @reaction_network SIR1 begin
     k2, I --> R
 end
 @test nameof(rn) == :SIR1
+
+### Tests some arrow variants ###
+
+rn1 = @reaction_network begin
+    @parameters a1 a2 k1 k2 b1
+    (a1, a2), C <--> 0
+    (k1, k2), A + B <--> C
+    b1, 0 <-- B
+end
+
+rn2 = @reaction_network begin
+    @parameters a1 a2 k1 k2 b1
+    a1, C --> 0
+    a2, 0 --> C
+    k1, A + B --> C
+    k2, C --> A + B
+    b1, B --> 0
+end
+
+ps = ones(5)
+u0 = [10.0, 20.0, 0.0]
+tspan = (0, 100.0)
+oprob1 = ODEProblem(rn1, u0, tspan, ps)
+oprob2 = ODEProblem(rn2, u0, tspan, ps)
+sol1 = solve(oprob1, Tsit5())
```

Comment:
Is this not covered by other existing tests?  This seems like a pretty basic example.

Why not just test equality of the two networks? Just give them the same name and then use `==`. That avoids having to add another ODE solve.